cmd/object-handlers.go

```diff
@@ -1158,7 +1179,11 @@ func (api objectAPIHandlers) PutObjectPartHandler(w http.ResponseWriter, r *http
 		}
 	}
 
-	partInfo, err := objectAPI.PutObjectPart(ctx, bucket, object, uploadID, partID, hashReader)
+	putObjectPart := objectAPI.PutObjectPart
+	if api.CacheAPI() != nil {
```

Comment:
don't cache if SSE header is set as well, we support multipart SSE-C as well remember.